src/startup.c

```diff
@@ -63,11 +69,53 @@ void dump_boot_args(struct boot_args *ba)
     printf("  machine_type: %d\n", ba->machine_type);
     printf("  devtree:      %p\n", ba->devtree);
     printf("  devtree_size: 0x%x\n", ba->devtree_size);
-    printf("  cmdline:      %s\n", ba->cmdline);
-    printf("  boot_flags:   0x%lx\n", ba->boot_flags);
-    printf("  mem_size_act: 0x%lx\n", ba->mem_size_actual);
+    int node = adt_path_offset(adt, "/chosen");
+
+    if (node < 0) {
+        printf("ADT: no /chosen found\n");
+        return;
+    }
+
+    u32 len;
+    const char *p = adt_getprop(adt, node, "firmware-version", &len);
+    u16 version = ba->revision;
+    if (strncmp(p, "iBoot-", 6) == 0) {
```

Comment:
We just added common code for this kind of test (for the Sequoia workaround), please use that. See firmware.c.

Reply:
I will see but I think I will need a separate fw_version_info because this is not macOS.

Reply:
Also in my case there's no guarantee of what versions the device may be on so I may have to add absolutely every release and beta...

Reply:
I think I will add a new function to the common code to directly compare iboot version.

Reply:
The iBoot version compare code does not rely on the firmware version table being exhaustive, it directly compares the iBoot version string. You only have to add the specific *threshold* versions to the table you want to test for, then anything in the range you specify will be considered as testing positive. I added it specifically because we need to test the iBoot1 version range (which we do not control) while we do control the subset of iBoot2 verisons we use.

Reply:
This particular piece of code is ran early, before `firmware_init()`, so this code still needs to parse iBoot version itself.